proposed/http-client/http-client-meta.md

```diff
@@ -53,9 +51,9 @@ a few drawbacks:
 
 ### Exceptions
 
-The domain exceptions `NetworkException`, `RequestException` and `HttpException` define
+The domain exceptions `NetworkException` and `RequestException` define
 a contract very similar to each other. The chosen approach is to not let them extend each other
-because inheritance does not make sense in the domain model. A `RequestException` is not a
+because inheritance does not make sense in the domain model. A `RequestException` is simply not a
 `NetworkException`.
```

Comment:
So one always needs two catch blocks?

Reply:
should we have a HttpClientExceptionInterface or something? i agree that an exception marking all exceptions of the PSR would be good design.

although in this case, the RequestException implies a programming error in the consumer (or possibly the client) or at least missing input validation. NetworkException can mean a misconfiguration (wrong server name) or a temporary network problem. RequestException are thus more likely to require attention from the development team. I might not want to log NetworkException even, but for example retry the action later.

Reply:
No, only one catch block is enough. There is the `ClientException` that "Every HTTP client related Exception MUST implement this interface."

Reply:
right, sorry, forgot about that. would it make sense to mention this here again? like "(However, both exceptions implement the ClientExceptionInterface if you want to generically catch exceptions while sending a request.)"